src/validation.cpp

```diff
@@ -1948,6 +1948,15 @@ bool CChainState::FlushStateToDisk(
                last_prune = std::max(1, std::min(last_prune, index.GetSummary().best_block_height));
             });
 
+            for (auto const& blocker : m_blockman.m_prune_blockers) {
+                const int blocker_height = blocker.second->nHeight - PRUNE_BLOCKER_BUFFER;
```

Comment:
In commit "Index: Use prune blockers for blockfilterindex" (7db7f26e97665c048b3f614c092899addd658f4f)

I'm seeing a segfault this line starting in the "Index: Use prune blockers" commit and continuing up to the ""Index: Skip pruning checks" commit in the `feature_blockfilterindex_prune.py` test because `blocker.second` is null. It would probably be good to not insert null prune blockers or just ignore them.

```
#0  0x000055555583d8a6 in CChainState::FlushStateToDisk (this=0x555555e46f10, state=..., mode=FlushStateMode::IF_NEEDED, nManualPruneHeight=0) at validation.cpp:1948
#1  0x0000555555849e51 in CChainState::ConnectTip (this=0x555555e46f10, state=..., pindexNew=0x7fffac001000, pblock=..., connectTrace=..., disconnectpool=...) at validation.cpp:2288
#2  0x000055555584f7f2 in CChainState::ActivateBestChainStep (this=0x555555e46f10, state=..., pindexMostWork=0x7fffac001000, pblock=std::shared_ptr<const CBlock> (empty) = {...}, fInvalidFound=@0x7fffa7ffe32f: false, connectTrace=...)
    at validation.cpp:2432
#3  0x000055555584fff8 in CChainState::ActivateBestChain (this=<optimized out>, state=..., pblock=std::shared_ptr<const CBlock> (empty) = {...}) at validation.cpp:2557
#4  0x00005555556c77fd in ThreadImport (chainman=..., vImportFiles=std::vector of length 0, capacity 0, args=...) at node/blockstorage.cpp:555
#5  0x0000555555633850 in operator() (__closure=0x7fffac000b60) at init.cpp:1656
#6  std::__invoke_impl<void, AppInitMain(NodeContext&, interfaces::BlockAndHeaderTipInfo*)::<lambda()>&> (__f=...) at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/bits/invoke.h:60
#7  std::__invoke_r<void, AppInitMain(NodeContext&, interfaces::BlockAndHeaderTipInfo*)::<lambda()>&> (__fn=...) at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/bits/invoke.h:110
#8  std::_Function_handler<void(), AppInitMain(NodeContext&, interfaces::BlockAndHeaderTipInfo*)::<lambda()> >::_M_invoke(const std::_Any_data &) (__functor=...)
    at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/bits/std_function.h:291
#9  0x0000555555b0ef5f in std::function<void ()>::operator()() const (this=0x7fffa7ffeb70) at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/bits/std_function.h:622
#10 util::TraceThread(char const*, std::function<void ()>) (thread_name=<optimized out>, thread_func=...) at util/thread.cpp:18
#11 0x000055555563376b in std::__invoke_impl<void, void (*)(char const*, std::function<void()>), char const*, AppInitMain(NodeContext&, interfaces::BlockAndHeaderTipInfo*)::<lambda()> > (
    __f=@0x555555e7f818: 0x555555b0edf0 <util::TraceThread(char const*, std::function<void ()>)>) at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/bits/invoke.h:60
#12 std::__invoke<void (*)(char const*, std::function<void()>), char const*, AppInitMain(NodeContext&, interfaces::BlockAndHeaderTipInfo*)::<lambda()> > (__fn=@0x555555e7f818: 0x555555b0edf0 <util::TraceThread(char const*, std::function<void ()>)>)
    at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/bits/invoke.h:95
#13 std::thread::_Invoker<std::tuple<void (*)(char const*, std::function<void()>), char const*, AppInitMain(NodeContext&, interfaces::BlockAndHeaderTipInfo*)::<lambda()> > >::_M_invoke<0, 1, 2> (this=0x555555e7f7e8)
    at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/thread:264
#14 std::thread::_Invoker<std::tuple<void (*)(char const*, std::function<void()>), char const*, AppInitMain(NodeContext&, interfaces::BlockAndHeaderTipInfo*)::<lambda()> > >::operator() (this=0x555555e7f7e8)
    at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/thread:271
#15 std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(char const*, std::function<void()>), char const*, AppInitMain(NodeContext&, interfaces::BlockAndHeaderTipInfo*)::<lambda()> > > >::_M_run(void) (this=0x555555e7f7e0)
    at /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0/include/c++/10.3.0/thread:215
#16 0x00007ffff7eaaca0 in ?? () from /nix/store/dzaadhn5y56da24635icafv1nrawxm4n-gcc-10.3.0-lib/lib/../lib64/libstdc++.so.6
#17 0x00007ffff7fb1e9e in start_thread () from /nix/store/gk42f59363p82rg2wv2mfy71jn5w4q4c-glibc-2.32-48/lib/libpthread.so.0
#18 0x00007ffff774949f in clone () from /nix/store/gk42f59363p82rg2wv2mfy71jn5w4q4c-glibc-2.32-48/lib/libc.so.6
```

Reply:
Oh, TIL that `Genesis()` can return a `nullptr` and does so initially when reindexing. I never really checked and assumed this couldn't happen since the Genesis block is hardcoded. Well, good to know and this should be fixed now.